packages/history-service/src/internal/history-multiplexer.ts

```diff
@@ -72,17 +73,25 @@ export class HistoryMultiplexer {
   public createNewRootLocationForMultipleConsumers(
     ...consumerLocations: ConsumerLocation[]
   ): RootLocationDescriptorObject {
-    let rootlocation: RootLocationDescriptorObject = {pathname: '/'};
+    let newRootLocation: RootLocationDescriptorObject = {pathname: '/'};
 
-    for (const consumerLocation of consumerLocations) {
-      rootlocation = this.createRootLocation(
-        consumerLocation.historyKey,
-        consumerLocation.location,
-        rootlocation
+    if (
+      this.rootLocationTransformer.createNewRootLocationForMultipleConsumers
+    ) {
+      newRootLocation = this.rootLocationTransformer.createNewRootLocationForMultipleConsumers(
+        ...consumerLocations
       );
+    } else {
+      for (const consumerLocation of consumerLocations) {
+        newRootLocation = this.rootLocationTransformer.createRootLocation(
+          newRootLocation,
+          consumerLocation.location,
+          consumerLocation.historyKey
+        );
+      }
     }
 
-    return rootlocation;
+    return this.setConsumerStates(newRootLocation, ...consumerLocations);
   }
 
   private createRootLocation(
```

Comment:
Shouldn't we also use `setConsumerStates` in this function?

Reply:
no, here we only add a single consumer state instead of setting/replacing all of them at once